### PR TITLE
Adds FSR sensors to SensorFilter

### PR DESCRIPTION
**Pull request by tayloryoung6396**
Needs testing and calibration, and probs another metric

### CODE REVIEW

**Comment by KipHamiltons** [comment]
This should probably rebase onto `hamiltons/examine_sensorfilter`

File: module/platform/darwin/SensorFilter/src/SensorFilter.cpp [hunk 1]
```diff
@@ -399,28 +410,70 @@ namespace module::platform::darwin {
...
+                                 && input.right.fsr3 < -config.footDown.certaintyFSRThreshold
+                                 && input.right.fsr4 < -config.footDown.certaintyFSRThreshold) {
+                            feet_down[BodySide::RIGHT] = false;
+                            feet_down[BodySide::LEFT]  = true;
+                        }
+                        // Right foot and left foot are roughly the same height in left foot space
+                        else {
+                            feet_down[BodySide::RIGHT] = true;
+                            feet_down[BodySide::LEFT]  = true;
+                        }
```
Comment (Bidski) [inline module/platform/darwin/SensorFilter/src/SensorFilter.cpp:454]: We used to calculate centre of pressure from the FSRs. Should be able to do something with centre of pressure being inside the foot's polygon or something?

Comment (Bidski) [reply module/platform/darwin/SensorFilter/src/SensorFilter.cpp:454]: https://github.com/NUbots/NUbots/blob/0063ef1a00874bcb790bc27c9d518c47c8b6c091/module/platform/darwin/SensorFilter/src/SensorFilter.cpp#L584-L588

https://github.com/NUbots/NUbots/blob/0063ef1a00874bcb790bc27c9d518c47c8b6c091/shared/utility/motion/ForwardKinematics.h#L556-L574

**Review by Bidski — CHANGES_REQUESTED** [review]

File: module/platform/darwin/SensorFilter/data/config/SensorFilter.yaml [hunk 2]
```diff
@@ -5,7 +5,9 @@ buttons:
 
 foot_down:
   from_load: true # If true uses the virtual load foot down, if false use the Z height method
-  certainty_threshold: 0.01 # Height difference of the Z for the Z height method
+  virtual_load: false # If true AND from_load is true uses the virtual load foot down, if false && from_load is true use fsr
```
[the remark this answers is not in the document]
Comment (tayloryoung6396) [reply module/platform/darwin/SensorFilter/data/config/SensorFilter.yaml:8]: Seems like effort, but totally would've something someone can do

File: module/platform/darwin/SensorFilter/src/SensorFilter.cpp (hunk 1 above)
Comment (tayloryoung6396) [reply module/platform/darwin/SensorFilter/src/SensorFilter.cpp:454]: Didn't know we knew the polygon, but thats a better way to do it. Will still need a threshold?

Comment (Bidski) [reply module/platform/darwin/SensorFilter/src/SensorFilter.cpp:454]: Didn't @MattAmos do something about feet polygons?

Comment (tayloryoung6396) [reply module/platform/darwin/SensorFilter/src/SensorFilter.cpp:454]: Is that in the codebase though? Or matlab?

Comment (Bidski) [reply module/platform/darwin/SensorFilter/src/SensorFilter.cpp:454]: Not sure

**Comment by KipHamiltons** [comment]
This branch shouldn't be forgotten. @tayloryoung6396

**Pull request closed without merging**